File: 2891-maximum-beauty-of-an-array-after-applying-operation/2891-maximum-beauty-of-an-array-after-applying-operation.java
class Solution {
    public int maximumBeauty(int[] nums, int k) {
        Arrays.sort(nums);
        int l = 0, r = 0, n = nums.length, ans = 0;

        while (r < n) {
            while (nums[r] - nums[l] > 2 * k) {
                l++;
            }
            ans = Math.max(ans, r - l + 1);
            r++;
        }

        return ans;
    }
}

File: 2891-maximum-beauty-of-an-array-after-applying-operation/2891-maximum-beauty-of-an-array-after-applying-operation.py
class Solution:
    def maximumBeauty(self, nums: List[int], k: int) -> int:
        nums.sort()
        l = r = 0
        n = len(nums)
        ans = 0

        while r < n:
            while nums[r] - nums[l] > 2 * k:
                l += 1
            ans = max(ans, r - l + 1)
            r += 1

        return ans
